stdlib/2/os/__init__.pyi

```diff
@@ -11,98 +15,220 @@ from mypy_extensions import NoReturn
+        # This is what the scandir interator yields
+        # The constructor is hidden
+
+        name: AnyStr
+        path: AnyStr
+        def inode(self) -> int: ...
+        def is_dir(self, follow_symlinks: bool = ...) -> bool: ...
+        def is_file(self, follow_symlinks: bool = ...) -> bool: ...
+        def is_symlink(self) -> bool: ...
+        def stat(self) -> stat_result: ...
+
+        def __fspath__(self) -> AnyStr: ...
+elif sys.version_info >= (3, 5):
+    class DirEntry(Generic[AnyStr]):
+        # This is what the scandir interator yields
+        # The constructor is hidden
+
+        name: AnyStr
+        path: AnyStr
+        def inode(self) -> int: ...
+        def is_dir(self, follow_symlinks: bool = ...) -> bool: ...
+        def is_file(self, follow_symlinks: bool = ...) -> bool: ...
+        def is_symlink(self) -> bool: ...
+        def stat(self) -> stat_result: ...
+
+
+class stat_result:
+    # For backward compatibility, the return value of stat() is also
+    # accessible as a tuple of at least 10 integers giving the most important
+    # (and portable) members of the stat structure, in the order st_mode,
+    # st_ino, st_dev, st_nlink, st_uid, st_gid, st_size, st_atime, st_mtime,
+    # st_ctime. More items may be added at the end by some implementations.
+
+    st_mode: int  # protection bits,
+    st_ino: int  # inode number,
+    st_dev: int  # device,
+    st_nlink: int  # number of hard links,
+    st_uid: int  # user id of owner,
+    st_gid: int  # group id of owner,
+    st_size: int  # size of file, in bytes,
+    st_atime: float  # time of most recent access,
+    st_mtime: float  # time of most recent content modification,
+    st_ctime: float  # platform dependent (time of most recent metadata change on Unix, or the time of creation on Windows)
+
+    if sys.version_info >= (3, 3):
+        st_atime_ns: int  # time of most recent access, in nanoseconds
+        st_mtime_ns: int  # time of most recent content modification in nanoseconds
+        st_ctime_ns: int  # platform dependent (time of most recent metadata change on Unix, or the time of creation on Windows) in nanoseconds
+
+    # not documented
+    def __init__(self, tuple: Tuple[int, ...]) -> None: ...
+
+    # On some Unix systems (such as Linux), the following attributes may also
+    # be available:
+    st_blocks: int  # number of blocks allocated for file
+    st_blksize: int  # filesystem blocksize
+    st_rdev: int  # type of device if an inode device
+    st_flags: int  # user defined flags for file
+
+    # On other Unix systems (such as FreeBSD), the following attributes may be
+    # available (but may be only filled out if root tries to use them):
+    st_gen: int  # file generation number
+    st_birthtime: int  # time of file creation
+
+    # On Mac OS systems, the following attributes may also be available:
+    st_rsize: int
+    st_creator: int
+    st_type: int
+
+class statvfs_result:  # Unix only
```

Comment:
This doesn't work -- the real `os.statvfs_result` class is also sequence of 10 integers. I found working code that breaks because of this. (And yes, this means that the PY3 stubs are also broken. Too bad. It still works.) Maybe for this a `NamedTuple` could work, even if it can't for `os.stat_result`. (I wonder if for the latter we could start with a namedtuple and then just subclass it and add a bunch more fields as properties or maybe just class attributes?)